vegafusion-python/vegafusion/runtime.py

```diff
@@ -218,9 +251,36 @@ def runtime(self) -> PyVegaFusionRuntime:
                 self.cache_capacity,
                 self.memory_limit,
                 self.worker_threads,
+                self._executor,
             )
         return self._runtime
 
+    @classmethod
+    def new_vendor(
+        cls,
+        vendor: Literal["datafusion", "sparksql"],
+        executor: PlanExecutor | None = None,
+        cache_capacity: int = 64,
+        memory_limit: int | None = None,
+        worker_threads: int | None = None,
+    ) -> VegaFusionRuntime:
+        from vegafusion._vegafusion import PyVegaFusionRuntime
```

Comment:
nit - throw the newly added vendor value doesn't match the expected values

```suggestion
        if vendor not in ("datafusion", "sparksql"):
            raise ValueError(f"Invalid vendor: {vendor}. Must be 'datafusion' or 'sparksql'")
        from vegafusion._vegafusion import PyVegaFusionRuntime
```